mobile-credential-online-presentation/README.md

```diff
@@ -0,0 +1,104 @@
+            "altText": "<YOUR_LOGO_ALT_TEXT>"
+        },
+        "headerText": "Verify your age before you continue",
+        "primaryColorHex": "#000000"
+    },
+    "resultAvailableInFrontChannel": true
+}'
+```
+
+### Making the app publically available
+
+The app needs to be publically available for the online presentation to work.
+You can either use a hosting service of your choice, or use cloudflare or ngrok tunnels to expose your locally running app publically.
+
+> [!IMPORTANT]
+> If you use a tunneling service, make sure to
+> * update the redirect URIs in your verifier tenant configuration (see above), and to
+> * access the app yourself via the public domain.
+>
+> The credential request to the verifier tenant validates the origin of the request and the request will fail when it is being made from `localhost`.
+
+### Credential holding in the wallet
+
+The default credential query requests a credential of doctype `org.iso.18013.5.1.mDL`, but you can change the credential query in the sample app directly.
+You need to ensure your wallet holds a credential that can fulfil the credential query you are using.
+
+
+## SDK Guide
+
+### Automatic mode detection
+
+If you don't provide a `mode` in the `RequestCredentialsOptions`, the SDK will determine the mode automatically by checking the user agent of the device in use.
+Mobile devices will use the same device flow, all other devices the cross device flow.
+
+### Same device presentation
+
+When request credentials in the same device flow, the user will be redirected to a wallet to fulfil the credential request.
+On completion of the request in the wallet, the user is redirected back to the browser to the redirect URI from the credential request.
+The `handleRedirectCallback` function from the SDK will extract the `response_code` from the URL, and retrieve the presentation results.
+
+> [!NOTE]
+> For the SDK to be able to retrieve the results on rendering the page the user is redirected to, the SDK needs to have the verifier tenant URL accessible.
+> The simplest way is to provide it as an environment variable when running the app.
+> Alternatively, you can persist the verifier tenant URL in `localStorage` when making the credential request, and retrieve it on the redirected tab.
+
+### Cross device presentation
+
+In the cross device flow, you don't need to invoke the `handleRedirectCallback`, but you can directly pass an `onComplete` and an `onFailure` handler to the `crossDeviceCallback` option in the credential request.
+
+## Running the app
+
+This is a NextJS app, using the App Router.
+You can configure your verifier tenant URL by setting the `NEXT_PUBLIC_API_BASE_URL` environment variable, e.g. by adding
+```
+NEXT_PUBLIC_API_BASE_URL=<YOUR_TENANT_URL>
+```
+to a `.env.local` file to the root of the project.
+
+This app only consists of a single page, and the main code on how to use the Web Verifier SDK can be found in `src/app/page.tsx`.
```

Comment:
```suggestion
This is a single page app. The main code using the Web Verifier SDK can be found in `src/app/page.tsx`.
```

Reply:
"single page app" (i.e. SPA) means something specific in this context. And this app is technically not a SPA. I'll improve the sentence similar to your suggestion, though.

Reply:
Fixed in https://github.com/mattrglobal/sample-apps/pull/167/commits/776c8c6240f228f1fbf937a92be937ecf460594f.